### DISCUSSION TITLE
HP z27 and z32 monitor wake-up issues

### DISCUSSION THREAD

**Discussion by jakedt** [MonitorControl/MonitorControl#594] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] I have read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

I was being bitten by #135 constantly before upgrading to 3.0.0. With 3.0.0, when resuming from sleep the external monitor won't wake up, and it won't be detected upon disconnect and reconnect. I had to un-plug the monitor to get it to be detected.

### Expected behavior

I would expect it to work like it does without monitor control: wake from sleep and continue to always be detected.

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.5.2
- Mac model: 13-inch, 2020, Four Thunderbolt 3 ports
- MonitorControl version: 3.0.0
- Monitor(s): HP z27
- Monitor Cable(s)/Connection(s): usb-c thunderbolt
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
Hi @jakedt

can you please clarify what do you mean?

1) The display itself stays in sleep mode (no desktop, etc), after waking up the mac or 
2) The display wakes up fine and you can use the desktop, but DDC control is not working and MonitorControl does not detect the display

If 1) is the case then it is most probably not a MonitorControl issue but a generic one (search for 'display does not wake macos' on google for various fixes) as the app does not interfere with system sleep or the wake up process in any way. It actually waits for about 5 seconds after sleep until it tries to redetect the display and allow for DDC control. 

What might be possible (looking at #135) is that the z27s have something wrong with their firmware and have issues with sleep/wake once DDC control was used - but it would be a weird problem and also a hardware one. You can try to set polling to 'None' in the advanced section of Displays and disable Hardware DDC control for a while to see if that changes things. If you consistently have the sleep problem only when DDC was used then this might be the problem.

**Reply by c08w38** [reply]
Hi there. It is the Problem you describe under 1.

I have the same issue with my Z27. The Issue is reproducible and it only occurs if MonitorControl is running. If i close the App, the Z27 goes to sleep and wakes up like it should...

**Reply by waydabber** [reply]
Does it happen if you set polling to 'None' or disable DDC altogether?

**Reply by c08w38** [reply]
Yes. It happen then too...
<img width="692" alt="CleanShot 2021-09-22 at 10 53 20" src="https://user-images.githubusercontent.com/4301492/134313453-a4845344-a0a2-4a6e-a058-fbe5e211af21.png">
<img width="694" alt="CleanShot 2021-09-22 at 10 53 33" src="https://user-images.githubusercontent.com/4301492/134313463-780d18a6-9082-4658-9182-60151c8b24c9.png">

**Reply by waydabber** [reply]
You can also try a [newer 4.0.0 build](https://www.dropbox.com/s/w1md5rjyll0lsyo/MonitorControl.zip?dl=1) to see if it helps as a lot of things have changed (note: this download is unsigned). However if DDC is disabled then it is difficult to see how MonitorControl can affect the sleeping ability of the display.

Huhh I see the german text clipping there. This should be corrected. :)

**Reply by c08w38** [reply]
In this version the Problem still exists if i leave DDC enabled. As soon as i disable DDC the problem is gone...

Still the same, if i kill the whole App, the Problem ist gone too... If it runs in the background, the Monitor wont wake up after sleep

**Reply by waydabber** [reply]
All right. This seems to be an issue with the z27 then I am afraid. I don't have such a display but maybe using ddcctl one could figure out what combination of commands make the display firmware bug like this. Or maybe there is a display setting/function that interferes with changing values via DDC (for example some settings in some registers are overwritten via DDC but not via display OSD or vice versa when there is a change and there might be some functions - maybe some image processing thing or something else - that rely on these settings changed via the OSD way). Maybe by tinkering with settings you might figure out what's wrong - if you find a solution please share it with others using this display. Also you could contact HP or look for a firmware update for the display (if it is upgradeable).

**Reply by c08w38** [reply]
Thank you for your support... i'm looking into it.

**Comment by waydabber** [comment]
Hi, I'll move this over to discussion as the issue seems to be sleeping + there is no much to do. Maybe other z27 owners will chime in as well.

**Comment by jakedt** [comment]
I haven't been able to repro yet. I will get more information if/when it happens again.

**Comment by dkocher** [comment]
This can still be reproduced with 4.1.0 Build 7034 with a HP Z27 Display. The workaround to uncheck _Preferences… → Displays → Use hardware DDC control_ resolves the problem.

**Comment by mravcak** [comment]
What worked for me on Z27 was setting _DCC read polling mode_ to _None_ in Preferences… → Displays.  Now it works as expected (i.e. the display still works after reconnecting).
